src/daemon/ca/publishing.rs

```diff
@@ -1,232 +1,1223 @@
+                        resource_class_name,
+                        updates,
+                    } => {
+                        objects.update_certs(resource_class_name, updates, timing, signer)?;
+                    }
+                    super::CaEvtDet::KeyPendingToActive {
+                        resource_class_name,
+                        current_key,
+                    } => {
+                        objects.add_class(resource_class_name, current_key, timing, signer)?;
+                    }
+                    super::CaEvtDet::KeyPendingToNew {
+                        resource_class_name,
+                        new_key,
+                    } => {
+                        objects.keyroll_stage(resource_class_name, new_key, timing, signer)?;
+                    }
+                    super::CaEvtDet::KeyRollActivated {
+                        resource_class_name, ..
+                    } => {
+                        objects.keyroll_activate(resource_class_name, timing, signer)?;
+                    }
+                    super::CaEvtDet::KeyRollFinished { resource_class_name } => {
+                        objects.keyroll_finish(resource_class_name)?;
+                    }
+                    super::CaEvtDet::CertificateReceived {
+                        resource_class_name,
+                        rcvd_cert,
+                        ..
+                    } => {
+                        // Update the received certificate if needed. If the URIs changed we may need to re-issue things
+                        objects.update_received_cert(resource_class_name, rcvd_cert)?;
+                        objects.re_issue_if_required(&self.config.issuance_timing, &self.signer)?;
+                    }
+                    super::CaEvtDet::ResourceClassRemoved {
+                        resource_class_name, ..
+                    } => {
+                        objects.remove_class(resource_class_name);
+                    }
+                    super::CaEvtDet::RepoUpdated { contact } => {
+                        objects.update_repo(contact);
+                    }
+                    _ => {}
+                }
+            }
+            Ok(())
+        })
+    }
 }
 
-//------------ ManifestInfo ------------------------------------------------
+impl CaObjectsStore {
+    fn key(ca: &Handle) -> KeyStoreKey {
+        KeyStoreKey::simple(format!("{}.json", ca))
+    }
+
+    fn cas(&self) -> KrillResult<Vec<Handle>> {
+        let cas = self
+            .store
+            .read()
+            .unwrap()
+            .keys(None, ".json")?
+            .iter()
+            .flat_map(|k| {
+                let mut res = None;
+                if let Some(name) = k.name().strip_suffix(".json") {
+                    if let Ok(handle) = Handle::from_str(name) {
+                        res = Some(handle)
+                    }
+                }
+                res
+            })
+            .collect();
```

Comment:
I can't quite follow what happens here as the docs for `flat_map()` say the closure should return an iterator like `.chars()` but here an `Option` is returned. Also, it feels like this is sort of filtering but it's unclear if the None values are included in the resulting collection, and if they are that feels a bit strange as the result is a list CA handles and Nones would presumablu not be interesting to the caller. So I was wondering if `filter_map()` would be a better fit here?